doc/cascadia/Command-Palette-spec.md

```diff
@@ -0,0 +1,269 @@
+commands added when the palette is opened. The way I have it now is as an
+overlay over the entire tab.
+
+The palette will consist of two main UI elements: a text box for searching for
+commands, and a list of commands.
+
+The list of commands will be populated with all the commands by default. Each
+command will appear like a `MenuFlyoutItem`, with an icon at the left (if it has
+one) and the name visible. When opened, the palette will automatically highlight
+the first entry in the list.
+
+The user can navigate the list of entries with the arrow keys. Hitting enter
+will close the palette and execute the action that's highlighted. Hitting escape
+will dismiss the pallete, returning control to the terminal. When the palette is
+closed for any reason (executing a command, dismissing with either escape or the
+`toggleCommandPalette` keybinding), we'll clear out any search text from the
+palette, so the user can start fresh again.
+
+We'll also want to enable the command palette to be filterable, so that the user
+can type the name of a command, and the command palette will automatically
+filter the list of commands. This should be more powerful then just a simple
+string compare - the user should be able to type a search string, and get all
+the commands that match a "fuzzy search" for that string. This will allow users
+to find the command they're looking for without needing to type the entire
+command.
+
+For example, consider the following list of commands:
+
+```json
+    "commands": [
+        { "icon": null, "name": "New Tab", "action": "newTab" },
+        { "icon": null, "name": "Close Tab", "action": "closeTab" },
+        { "icon": null, "name": "Close Pane", "action": "closePane" },
+        { "icon": null, "name": "[-] Split Horizontal", "action": "splitHorizontal" },
+        { "icon": null, "name": "[ | ] Split Vertical", "action": "splitVertical" },
+        { "icon": null, "name": "Next Tab", "action": "nextTab" },
+        { "icon": null, "name": "Prev Tab", "action": "prevTab" },
+        { "icon": null, "name": "Open Settings", "action": "openSettings" },
+        { "icon": null, "name": "Open Media Controls", "action": "openTestPane" }
+    ],
+```
+
+* "open" should return both "**Open** Settings" and "**Open** Media Controls".
+* "Tab" would return "New **Tab**", "Close **Tab**", "Next **Tab**" and "Prev
+  **Tab**".
+* "P" would return "Close **P**ane", "[-] S**p**lit Horizontal", "[ | ]
+  S**p**lit Vertical", "**P**rev Tab", "O**p**en Settings" and "O**p**en Media
+  Controls".
+* Even more powerfully, "sv" would return "[ | ] **S**plit **V**ertical". This
+  is a great example of how a user could execute a command with very few
+  keystrokes.
+
+As the user types, we should **bold** each matching character in the command
+name, to show how their input correlates to the results on screen.
+
+## Capabilities
+
+### Accessibility
+
+As the entire command palette will be a native XAML element, it'll automatically

```

Comment:
Some comments regarding accessibility to just be aware of:
- **XAML Focus**: we definitely need the focus to shift properly. That'll help a lot with accessibility.
- **Navigation w/ Terminal**: you won't be able to interact with the terminal while this command palette is open, right? Otherwise, we might have to think about how navigating the UIA tree _should_ look like for this element (even though XAML does some of that for us).

Reply:
I'll make sure to call those out.